sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -282,6 +282,8 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
         builder.append("]")
         builder.build()
       })
+    case udt: UserDefinedType[_] =>
```

Comment:
Is this the only place we miss UDT in Cast?

Reply:
what about cast UDT to other types?

Reply:
ok, I'll check

Reply:
I think we can strip UDT at L608 and fix this problem entirely.

Reply:
Here?
https://github.com/apache/spark/blob/cd9f49a2aed3799964976ead06080a0f7044a0c3/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L608

Since we can cast UDTs to other UDTs or strings only, IIUC there is only a place to fix?

Reply:
i see, makes sense